WordPress/src/main/java/org/wordpress/android/util/PerAppLocaleManager.kt

```diff
@@ -118,7 +124,30 @@ class PerAppLocaleManager @Inject constructor(
         }
     }
 
-    companion object {
-        const val EXPERIMENTAL_PER_APP_LANGUAGE_PREF_KEY = "experimental_per_app_language_prefs"
+    /**
+     * Called when the device language is changed from our in-app language picker
+     * TODO Detect when language changed from app settings dialog
+     */
+    fun onLanguageChanged(languageCode: String) {
+        if (languageCode.isEmpty()) {
+            return
+        }
+
+        // Only update if the language is different
+        if (languageCode.equals(getCurrentLocaleLanguageCode()).not()) {
+            setCurrentLocaleByLanguageCode(languageCode)
+        }
+
+        // Track language change on Analytics because we have both the device language and app selected language
+        // data in Tracks metadata.
+        val properties: MutableMap<String, Any?> = HashMap()
+        properties["app_locale"] = languageCode
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ACCOUNT_SETTINGS_LANGUAGE_CHANGED, properties)
+
+        // Language is now part of metadata, so we need to refresh them
+        AnalyticsUtils.refreshMetadata(accountStore, siteStore)
+
```

Comment:
Question (❓): During transition, this below logic got completely removed from this function, can you help me understand why? 🙏 

```
        // Refresh the app
        Intent refresh = new Intent(getActivity(), getActivity().getClass());
        startActivity(refresh);
        getActivity().setResult(LANGUAGE_CHANGED);
        getActivity().overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        getActivity().finish();
```

Reply:
It's no longer necessary since the system now handles the language change. I can probably get rid of the `LANGUAGE_CHANGED` const entirely - I'll do that in a subsequent PR.